packages/components/src/internal/components/base/ViewAsToggle.tsx

```diff
@@ -0,0 +1,40 @@
+import React, { PureComponent, ReactNode } from 'react';
+import { Map } from 'immutable';
+
+interface Props {
+    selected: string;
+    options: Map<string, string>;
+    onSelectionChange: (selected: string) => void;
+    id?: string;
+}
+
+export class ViewAsToggle extends PureComponent<Props> {
```

Comment:
The only thing that makes this a "View" toggle is the label "View".  I wonder if it's worth making that text a parameter and changing the name to be something more general.

Reply:
Sure. Done.